components/pkg-export-docker/src/build.rs

```diff
@@ -82,6 +104,8 @@ pub struct BuildSpec<'a> {
     pub idents_or_archives: Vec<&'a str>,
     /// The Builder Auth Token to use in the request
     pub auth: Option<&'a str>,
+    /// Base image used in From of dockerfile
+    pub base_image: Option<&'a str>,
```

Comment:
The more work I do in rust, the more I've found `Option`s inside types (as opposed to parameters or return values) to be an anti-pattern.

In this case, I'd suggest
```suggestion
    pub base_image: String,
```
This allows `docker_base_image` to be simplified to:
```rust
fn default_docker_base_image() -> String {
    #[cfg(unix)]
    {
        DEFAULT_BASE_IMAGE.to_string()
    }

    #[cfg(windows)]
    {
        format!("{}:{}",
                DEFAULT_BASE_IMAGE,
                docker::default_base_tag_for_host()?)
    }
}
```
(And perhaps inlining those `DEFAULT_BASE_IMAGE` constants since they're not used anywhere else and to make it clear they're different across the different platforms would be even better)

With that change, the handling of the optional case can go _before_ the `BuildSpec` is created in `new_from_cli_matches`:
```rust
                    base_image:         m.value_of("BASE_IMAGE")
                                         .map(str::to_string)
                                         .unwrap_or_else(default_docker_base_image), }
```